src/msg/simple/Pipe.cc

```diff
@@ -1424,6 +1424,8 @@ void Pipe::discard_requeued_up_to(uint64_t seq)
     rq.pop_front();
     out_seq++;
   }
+
+  out_seq = seq;
```

Comment:
if out_q is empty, this method would be a no-op, as it returns at https://github.com/ceph/ceph/pull/21585/files#diff-459b1eb43a0f5b6b6b430e3c67fee6ceR1414 . so, in which case, your change kicks in?